include/xsimd/math/xsimd_scalar.hpp

```diff
@@ -0,0 +1,292 @@
+        return std::isnan(x - x) ? std::numeric_limits<T>::NaN() : x - std::trunc(x);
+    }
+
+    template <class T>
+    inline bool is_odd(const T& x)
+    {
+        return is_even(x - 1.);
+    }
+
+    template <class T>
+    inline bool is_even(const T& x)
+    {
+        return is_flint(x * 0.5);
+    }
+
+    template <class T>
+    inline T exp10(const T& x)
+    {
+        // FIXME: faster alternatives exist
+        return std::pow(T(10), x);
+    }
+
+    template <class T>
+    inline std::complex<T> expm1(const std::complex<T>& val)
+    {
+        // FIXME: probably not accurate
+        return std::exp(val) - 1;
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, T0>::type
+    pow(const T0& t0, const T1& t1)
+    {
+      return detail::ipow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_floating_point<T1>::value, T0>::type
+    pow(const T0& t0, const T1& t1)
+    {
+      return std::pow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<std::is_integral<T1>::value, std::complex<T0>>::type
+    pow(const std::complex<T0>& t0, const T1& t1)
+    {
+      return detail::ipow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline typename std::enable_if<!std::is_integral<T1>::value, std::complex<T0>>::type
+    pow(const std::complex<T0>& t0, const T1& t1)
+    {
+      return std::pow(t0, t1);
+    }
+
+    template <class T0, class T1>
+    inline auto
+    pow(const T0& t0, const std::complex<T1>& t1)
+    -> typename std::enable_if<std::is_scalar<T0>::value, decltype(std::pow(t0, t1))>::type
+    {
+      return std::pow(t0, t1);
+    }
+
+    template<class T>
+    inline auto bitofsign(T const& x) -> decltype(std::signbit(x))
+    {
+        return std::signbit(x);
+    }
+
+    template <class T>
+    inline auto signbit(T const& v) -> decltype(bitofsign(v))
+    {
+        return bitofsign(v);
+    }
+
+    inline double sign(bool const &v)
+    {
+        return v;
+    }
+
+    template <class T>
+    inline double sign(T const &v)
+    {
+        return v < 0 ? -1. : v == 0 ? 0. : 1.;
+    }
+
+    template <class T>
+    inline std::complex<T> sign(const std::complex<T>& v)
+    {
+        if (v.real())
+            return {sign(v.real()), 0};
+        else
+            return {sign(v.imag()), 0};
+    }
+
+    template <class T>
+    std::complex<T> log2(const std::complex<T>& val)
+    {
+        return std::log(val) / std::log(2);
+    }
+
+    template <class T>
+    inline std::complex<T> log1p(const std::complex<T>& val)
+    {
+        return std::log(val + 1);
+    }
+
+
+    template <class T0, class T1>
+    inline auto min(T0 const &self, T1 const &other) ->
+        typename std::enable_if<std::is_scalar<T0>::value && std::is_scalar<T1>::value,
+                                typename std::decay<decltype(self > other ? other : self)>::type>::type
```

Comment:
`enable_if_t` and `decay_t`

Reply:
These are c++14 while only c++11 is required for now in the cmakelists.txt. Let's change that in a dedicated PR.

Reply:
Yep. I thought xsimd was c++11, then checked the docs — they state it's c++14. No problem either way, we might want to adjust the docs here https://xsimd.readthedocs.io/en/latest/